main.js

```diff
@@ -0,0 +1,226 @@
+// displayMap function handles displaying the 
+// pin on a particular restaurant
+async function displayMap(restaurant) {
+  // get the address of the restaurant passed into displayMap
+  const address = await getAddress(restaurant);
+  const website = await getWebsite(restaurant);
+  console.log({restaurant});
+  console.log({address});
+
+  // fetch the json data from nomatim API for the address requested
+  fetch(`https://nominatim.openstreetmap.org/search/?q=${address}&format=json`)
+  .then(response => response.json())  // promise response to json
+  .then(json => {
+    console.log({lat: json[0].lat, lon: json[0].lon});
+    // assing the map variable, set it to id of html div
+    // and set the view to lat and lon of restaurant
+    var map = L.map('mapid').setView([json[0].lat, json[0].lon], 16);
+  
+    // add tile layer to map, from Interactive Maps slides, change later?
+    L.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
+      attribution: 'Map data &copy; <a href="https://www.openstreetmap.org/">OpenStreetMap</a> contributors'
+    }).addTo(map);
+
+    // if there is an address passed into the function
+    if(address !== 'Burlington, VT 05401') {
+      // set marker on the coordinates of the restaurant
+      L.marker([json[0].lat, json[0].lon]).addTo(map)
+      .on('click', () => document.location.href = website);
+        
+    // otherwise, address is Burlington so display all the markers
+    } else {
+      // pass map so function knows where to put markers
+      displayAllMarkers(map);
+    }
+  });
+}
+
+// takes the name of a restaurant and fetches the json
+// file for that restaurant, 
```

Comment:
Good use of comments. I like how you've separated out the distinct actions into their own functions.